h5py/_conv.pyx

```diff
@@ -121,12 +132,72 @@ cdef herr_t init_generic(hid_t src, hid_t dst, void** priv) except -1:
     priv[0] = sizes
     sizes[0].src_size = H5Tget_size(src)
     sizes[0].dst_size = H5Tget_size(dst)
+    log_convert_registered(src, dst)
 
     return 0
 
 # =============================================================================
 # Vlen string conversion
 
+_H5PY_PY_TAG = str(H5PY_PYTHON_OPAQUE_TAG)  # cache this
+cdef bint _is_pyobject_opaque(hid_t obj):
+    # This complexity is needed to sure:
+    #   1) That ctag is freed
+    #   2) We don't segfault (for some reason a try-finally statement is needed,
+    #   even if we do (what I think are) the right steps in copying and freeing.
+    cdef char* ctag = NULL
+    try:
+        if H5Tget_class(obj) == H5T_OPAQUE:
+            ctag = H5Tget_tag(obj)
+            if ctag != NULL:
+                tag = str(ctag)
+                if tag == _H5PY_PY_TAG:
+                    return True
+        return False
+    finally:
+        IF HDF5_VERSION >= (1, 8, 13):
```

Comment:
Do we have a minimum 1.8.x version we support? It [looks like](https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/) 1.8.13 was released in 2014, so I'd say we could reasonably ignore older versions by now.

Reply:
I copied this from https://github.com/h5py/h5py/blob/master/h5py/h5t.pyx#L635, after trying all the if related experiments. We could do a overall cleanup for HDF5 versions at some point (like we did for py3).